src/main.rs

```diff
@@ -1,3 +1,13 @@
+extern crate core;
+
+use crate::CommsError::{
+    ConnectionClosed, ConnectionExists, ConnectionNotFound, ServerLimitReached, UnexpectedHandshake,
+};
+use crate::Connection::Closed;
+use crate::Connection::Open;
+use crate::MessageType::{GetCount, Handshake, Post};
+use crate::Response::{HandshakeReceived, PostReceived};
+use std::borrow::Borrow;
```

Comment:
Zbędne importy. Jeśli chodziło o używanie nazw bez ich namespace'ów to i tak nie korzystasz z tego konsekwentnie w kodzie.